components/popover/popover-mixin.js

```diff
@@ -102,16 +175,31 @@ export const PopoverMixin = superclass => class extends superclass {
 		if (this._useNativePopover) this.hidePopover();
 
 		this._previousFocusableAncestor = null;
-		this._removeAutoCloseHandlers();
-		this._clearDismissible();
+		this.#removeAutoCloseHandlers();
+		this.#removeRepositionHandlers();
+		this.#clearDismissible();
 		await this.updateComplete; // wait before applying focus to opener
-		this._focusOpener();
+		this.#focusOpener();
 		this.dispatchEvent(new CustomEvent('d2l-popover-close', { bubbles: true, composed: true }));
+
 	}
 
 	configure(properties) {
+		this._margin = properties?.margin ?? 18;
+		this._maxHeight = properties?.maxHeight;
+		this._maxWidth = properties?.maxWidth;
+		this._minHeight = properties?.minHeight;
+		this._minWidth = properties?.minWidth;
 		this._noAutoClose = properties?.noAutoClose ?? false;
+		this._noAutoFit = properties?.noAutoFit ?? false;
 		this._noAutoFocus = properties?.noAutoFocus ?? false;
+		this._noPointer = properties?.noPointer ?? false;
+		this._offset = properties?.offset ?? 16;
+		this._preferredPosition = {
```

Comment:
Not sure how often we're expecting `configure` to be called, but this line is going to result in a re-render 100% of the time, even when location/span/allowFlip don't change.

Reply:
That's a good point. The way I have the test consumer set up, `configure` should only get called when one of these properties has changed and a re-render _will_ be necessary. While that's how I'd expect consumers to be set up, it's entirely possible for them to unnecessarily call `configure`, and we can probably guard against that here.

Reply:
Ok, I think I've addressed this. If no properties are passed in then we use default "preferred position" which will always be the same object instance. If properties are passed and any of these are different, we set a new "preferred position" object.